oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserPrincipalProvider.java

```diff
@@ -287,7 +287,16 @@ private Set<Principal> getGroupMembership(@NotNull Tree authorizableTree) {
 
             // remember the regular groups in case caching is enabled
             if (doCache) {
-                cacheGroups(authorizableTree, groupPrincipals);
+                PrincipalCommitterThread commitThread = UserPrincipalProviderCommitterProvider.getInstance().cacheGroups(authorizableTree, groupPrincipals, expiration, root);
```

Comment:
the commitThread could just be a Thread object. it the userprincipalprovider can be ignorant about the implementation, no?